apiserver/authentication/jwt/doc.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// Package jwt provides an authentication mechanism whereby a JAAS-registered
+// Juju controller can accept a JAAS JWT (JSON Web Token) as the means to
+// declare a user and their permissions.
+//
+// This mechanism kicks in if, after setting up the JWT in JAAS, you bootstrap your
+// controller with the 'login-token-refresh-url' controller config set to
+// the JAAS JWT refresh endpoint.
+//
+// Juju uses a [github.com/juju/juju/apiserver/authentication.PermissionDelegator]
+// instance to look up what permissions a user has. For non-JWT scenarios, these
+// come from the Juju permissions model. However, when a JWT is used, the Juju
+// permission model is empty, and the permissions are retrieved from the JWT.
+// JAAS provides a finer-grained permission model than Juju, and so constructs the
+// JWT to contain permissions matching those which Juju understands.
+//
+// Configuring a Juju controller to use a JWT for authentication means that the
+// discharge of cross model relation macaroons is performed by JAAS, which uses
+// its permissions model to validate access for consuming offers.
+
```

Comment:
Is that the only way in which a JWT affects things on the Juju side (i.e., are these all cases where Juju will defer to JAAS for authentication)? (Your original draft suggested there was more. That's what my botched attempt at itemizing them was about.)

Reply:
Oh I see, yes. But your original rework conflated authentication and authorisation. I've added a sentence to the first paragraph to cover the authentication aspects again.

Reply:
I was thinking of both in terms of impact. As in, what changes in Juju as a result of configuring your controller with a JWT? Both the authentication and the authorisation. Basically I don't like how the two are now separated. (Plus I'd spell "http" and "api" with capitals.)

Reply:
They are separate concerns and IMO don't belong together. Authentication happens up front as part of login. Authorisation happens downstream from that in disconnected workflows. They are typically performed by completely separate subsystems. eg candid for authentication, juju permission model for authorisation. Maybe easier to discuss f2f.